tee-worker/sidechain/rpc-handler/src/direct_top_pool_api.rs

```diff
@@ -333,27 +340,59 @@ where
 	response.map_err(|e| format!("{:?}", e))
 }
 
-async fn request_vc_inner(params: Params) -> Result<RpcReturnValue, String> {
+fn author_submit_request_vc_inner<R, TCS, G>(author: Arc<R>, params: Params) -> Result<H256, String>
+where
+	R: AuthorApi<H256, H256, TCS, G> + Send + Sync + 'static,
+	TCS: PartialEq + Encode + Decode + Debug + Send + Sync + 'static,
+	G: PartialEq + Encode + Decode + Debug + Send + Sync + 'static,
+{
 	let payload = get_request_payload(params)?;
-	let request = AesRequest::from_hex(&payload)
-		.map_err(|e| format!("AesRequest construction error: {:?}", e))?;
+	let request = AesRequest::from_hex(&payload).map_err(|e| format!("{:?}", e))?;
+
+	// Use this hash to trigger the storage of connection_registry
+	let hash = request.using_encoded(|x| BlakeTwo256::hash(x));
+
+	thread::spawn(move || request_vc_inner1(author, request, hash));
```

Comment:
As mentioned, although unlikely - it's still better to make sure this thread only runs after the connection registry has been populated - can be followed up by another issue